src/core/src/core_logic/PatchInstaller.py

```diff
@@ -800,4 +810,20 @@ def get_max_batch_size(self, maintenance_window, package_manager):
         self.composite_logger.log_debug("Calculated max batch size is: {0}".format(max_batch_size_for_packages))
 
         return max_batch_size_for_packages
-
+    
+    def is_install_patches_in_single_batch_enabled(self):
+        """"""
+        if self.install_patches_in_single_batch == True:
+            return self.install_patches_in_single_batch
+        
+        dur = datetime.datetime.strptime(self.execution_config.duration, "%H:%M:%S")
+        dura = timedelta(hours=dur.hour, minutes=dur.minute, seconds=dur.second)
+        total_time_in_minutes = self.env_layer.datetime.total_minutes_from_time_delta(dura)
+        max_maintenance_window_criteria = total_time_in_minutes >= Constants.MAX_PATCH_OPERATION_DURATION_IN_MINUTES
+        no_inclusion_exclusion_criteria = not (self.package_filter.is_exclusion_list_present() or self.package_filter.is_inclusion_list_present())
+        classification_criteria = self.package_filter.is_msft_critsec_classification_only()       
+        suse_criteria = self.env_layer.get_package_manager() == Constants.ZYPPER
+        user_flag_criteria = os.path.exists(Constants.INSTALL_PATCHES_IN_SINGLE_BATCH)
```

Comment:
what if os.path.exists returns false, what check you do you have to ensure user_flag_criteria is true becuz on 827 it looks like it must be true, same goes for line 824-826, maybe wrap them inside a try exception, so we catch any exception